File: sdk/java/src/main/java/com/anyvali/parse/Defaults.java
package com.anyvali.parse;

import com.anyvali.Schema;

/**
 * Default value application utilities.
 */
public final class Defaults {
    private Defaults() {}

    /**
     * Deep copy a default value before applying it.
     */
    public static Object applyDefault(Object defaultValue) {
        return Schema.deepCopyValue(defaultValue);
    }
}

File: sdk/kotlin/src/main/kotlin/com/anyvali/parse/Defaults.kt
package com.anyvali.parse

object Defaults {
    /**
     * Apply a default value if the input is absent (not present).
     * null is NOT treated as absent.
     */
    fun applyDefault(present: Boolean, inputValue: Any?, defaultValue: Any?): Any? {
        return if (present) inputValue else defaultValue
    }
}
